### PR TITLE
Port to 1.20.1

### PR DESCRIPTION
**Pull request by Mrbysco**
This PR ports the mod to 1.20.1

Happy ModToberFest :)

### CODE REVIEW

File: src/main/java/knightminer/simplytea/data/gen/RecipeGenerator.java [hunk 1]
```diff
@@ -116,10 +111,10 @@ protected void buildCraftingRecipes(Consumer<FinishedRecipe> consumer) {
 
 		// advanced tea
 		addTea(consumer, cup_cocoa, Items.COCOA_BEANS, Items.COCOA_BEANS, teapot_frothed);
-		addHoney(consumer, cup_cocoa, tea_stick, CocoaItem.CINNAMON_TAG);
+		addHoney(consumer, RecipeCategory.MISC, cup_cocoa, tea_stick, CocoaItem.CINNAMON_TAG);
```
Comment (KnightMiner) [inline src/main/java/knightminer/simplytea/data/gen/RecipeGenerator.java]: I feel like most of the tea crafting recipes should be under food.

The pots themselves I think misc is reasonable, I could go either way on the tea bags and water boiling and alike as long as its consistent. But anything that is edible I'd put under food.

File: src/main/java/knightminer/simplytea/data/gen/ShapelessHoneyRecipe.java [hunk 2]
```diff
@@ -31,7 +33,7 @@ public class ShapelessHoneyRecipe extends ShapelessRecipe {
 	private final Ingredient honey;
 	private final String tag;
 	public ShapelessHoneyRecipe(ResourceLocation id, String group, ItemLike tea, Ingredient honey, String tag) {
-		super(id, group, TeaCupItem.withHoney(new ItemStack(tea), tag), NonNullList.of(Ingredient.EMPTY, Ingredient.of(tea), honey));
+		super(id, group, CraftingBookCategory.MISC, TeaCupItem.withHoney(new ItemStack(tea), tag), NonNullList.of(Ingredient.EMPTY, Ingredient.of(tea), honey));
```
Comment (KnightMiner) [inline src/main/java/knightminer/simplytea/data/gen/ShapelessHoneyRecipe.java:36]: Food also makes more sense for this category; arguably should be just passed in from JSON however shapeless recipes do it if thats not too much effort.

File: src/main/java/knightminer/simplytea/data/gen/WorldgenGenerator.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import net.minecraftforge.registries.ForgeRegistries;
+
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+public class WorldgenGenerator extends DatapackBuiltinEntriesProvider {
+    public static final RegistrySetBuilder BUILDER = new RegistrySetBuilder()
+            .add(Registries.CONFIGURED_FEATURE, Registration::configuredFeatures)
+            .add(Registries.PLACED_FEATURE, Registration::placedFeatures)
+            .add(ForgeRegistries.Keys.BIOME_MODIFIERS, Registration::biomeModifiers);
```
Comment (KnightMiner) [inline src/main/java/knightminer/simplytea/data/gen/WorldgenGenerator.java]: Any reason these need to be defined in registration over just methods in this class? I like to keep code only used in datagen in separate classes when reasonable.

**Comment by KnightMiner** [comment]
Mostly looks good, just a couple of minor comments.

File: src/main/java/knightminer/simplytea/data/gen/ShapelessHoneyRecipe.java (hunk 2 above)
Comment (Mrbysco) [reply src/main/java/knightminer/simplytea/data/gen/ShapelessHoneyRecipe.java:36]: CraftingBookCategory only has 4 values: BUILDING, REDSTONE, EQUIPMENT and MISC

File: src/main/java/knightminer/simplytea/data/gen/WorldgenGenerator.java (hunk 3 above)
Comment (Mrbysco) [reply src/main/java/knightminer/simplytea/data/gen/WorldgenGenerator.java]: No reason, didn't know if you wanted them in the same class or not

**Pull request merged**